docs/antrea-network-policy.md

```diff
@@ -948,6 +955,71 @@ spec:
 The above example allows all Pods from Namespace "default" to connect to all "core-dns"
 Pods from Namespace "kube-system" on TCP port 53.
 
+## FQDN based filtering
+
+Antrea-native policy accepts a `fqdn` field to restrict egress access to the Fully
+Qualified Domain Names (FQDN) specified by exact name or wildcard matching patterns.
+
+The standard `Allow`, `Drop` and `Reject` actions apply to FQDN egress rules.
+
+An example FQDN based filtering could look like this:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: cnp-fqdn-all-google
+spec:
+  priority: 1
+  appliedTo:
+  - podSelector:
+      matchLabels:
+        app: client
+  egress:
+  - action: Drop
+    to:
+      - fqdn: "*google.com"
+```
+The above example drops all traffic destined to any FQDN that matches the wildchard
+expression `*google.com` originating from any Pod with `app` set to `client` across
+any Namespace. This feature only works at the L3/L4 level. DNS response packets will
+only be dropped if the client's requested FQDN is matched by a FQDN policy rule but
+the datapath reconciliation for that rule fails.
```

Comment:
The last sentence comes a bit out of nowhere here and seems to be about a detail of the implementation. I am not sure it fits within the rest of the documentation since we do not explain anywhere (yet) how this feature is implemented with respect to DNS queries.